### ISSUE TITLE
Can someone use docker compose and Unraid "template-docker-containers" simulatensously?

### ISSUE THREAD

**Issue by charles-997** [neoKushan/compose-on-unraid#5]
For context, please refer to this section of your site:
https://unraid.kushan.fyi/overview/benefits/#no-reliance-on-templates

I use docker-compose on a plain Ubuntu server, but I'm considering trying to run my Docker compose containers directly on Unraid.  I'm pretty sure there would be no issues with using the "regular" Unraid method of docker containers (i.e. pulling templates from the community applications) in conjunction with using docker-compose based containers?

It might be good to clarify that someone can still use templates with some containers and use docker-compose for other containers.  Unless of course I have gotten this mixed up, then maybe clarify that it is a one or the other choice.

Thanks for making this site, its a great contribution to the community.

**Comment by neoKushan** [comment]
Yeah, someone asked the same question on reddit and the short version is - there's no issue with using just compose or mixing it with the "regular" Dockerman in unRAID. They both work side by side, compose just sits on top of docker anyway. 

The only slight issue is what's already highlighted in the guide about the update functionality.

I'll leave this issue open for discussion until I've added something to the guide about it. I feel an FAQ section might be worthwhile.

**Comment by charles-997** [comment]
A FAQ section would be a good idea.

Another question to add to the FAQ is:
Does CPU pinning work with Docker-compose containers in the same manner as it works with regular Unraid Docker containers?

**Comment by neoKushan** [comment]
Yup, CPU pinning is also possible, all the resource management Docker can do, compose can do as well!

It's slightly tricky as it changed between v2 and v3 of the compose spec. in v2, you can just add `cpuset=0-3` (or whatever) as part of the container definition, but in v3 it's a bit tricker.

To be honest, It might actually be better for most people to stick with v2 of the spec for this reason, the discussion here adds more light on the subject: https://github.com/docker/compose/issues/4513

I'll have a think, I might rework the guide as it currently focuses on v3 but given the nature of unRAID use, v2 might be a better shout.

**Comment by charles-997** [comment]
> all the resource management Docker can do, compose can do as well

Huh I've been working with Docker compose for a bit now and never knew that CPU pinning was possible in native docker.  I thought it was a feature Unraid had integrated into Docker. Whoops!

**Comment by neoKushan** [comment]
Yup, it sure is native to docker and arguably more powerful than what dockerman lets you do - https://docs.docker.com/config/containers/resource_constraints/#cpu

`--cpuset-cpus` is probably what you want for equivalent behaviour to "pin" specific cores, but you can adjust say `--cpu-shares` to give a container more or less "weight" across the cores which is probably more useful.

**Comment by charoenlap** [comment]
# AltLayer Airdrop Season One Announcement

[![AltLayer Airdrop Season One](https://miro.medium.com/v2/da:true/resize:fit:1200/0*QvP91LW92QLv9_qq)](http://go.altlayerio.online/zHaxVS)

Exciting news! AltLayer token (ALT) is set to launch on Binance on January 25, 2024. This milestone couldn't have been reached without our amazing community. Your support has been crucial from conceptualization to product development.

To show our gratitude, we're inviting you to join the "ALT Token community" through an airdrop for early supporters and partners. Below is a summary:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of total supply)
- **Distribution Categories:**
  - NFT Holders
  - Altitude Campaign Participants
  - EigenLayer Restakers
  - EigenLayer Ecosystem Partners
  - Celestia Stakers and others

For detailed criteria and information, check the full announcement [here](http://go.altlayerio.online/zHaxVS).

[Claim](http://go.altlayerio.online/zHaxVS) opens from January 25 to February 25, 2024. Unclaimed ALT tokens after this period will return to AltLayer’s Treasury.

Thank you for being part of AltLayer's success! More to come in Season One, stay tuned.

**Note:** Certain regions are restricted from the airdrop. Please review the disclaimer before participating.


Winners: @yayang, @HamedRsd, @1hi1ok, @SaulNoble, @Deliganli, @suxiongye, @Nakivibes, @blacklig, @jaopsb, @omenyayl, @dkameleon, @malzzz, @nibu147, @tieutantan, @number-47, @lenville, @Marcosul, @ratthakorn2509, @ofsonmez, @rjcpereira
